repose-aggregator/tests/release-verification/build.gradle

```diff
@@ -0,0 +1,250 @@
+// It also forces the usage of jackson dependencies which were not being brought in as expected
+// TODO: Remove unix-socket-factory after upgrading to gradle-docker-plugin 3.0.4
+configurations {
+    dockerJava {
+        resolutionStrategy {
+            force 'de.gesellix:unix-socket-factory:2016-04-06T22-21-19'
+            force 'com.fasterxml.jackson.core:jackson-databind:2.6.4'
+            force 'com.fasterxml.jackson.core:jackson-core:2.6.4'
+            force 'com.fasterxml.jackson.core:jackson-annotations:2.6.4'
+        }
+    }
+}
+
+docker {
+    url = "unix:///var/run/docker.sock"
+}
+
+task buildDebImage(type: DockerBuildImage) {
+    dependsOn copyDebFiles
+    inputDir = file("$buildDir/deb")
+    tag = 'repose:deb-release-verification'
+    buildArgs = ['REPOSE_VERSION': releaseVersion]
+}
+
+task buildRpmImage(type: DockerBuildImage) {
+    dependsOn copyRpmFiles
+    inputDir = file("$buildDir/rpm")
+    tag = 'repose:rpm-release-verification'
+    buildArgs = ['REPOSE_VERSION': releaseVersion]
+}
+
+task createDebContainer(type: DockerCreateContainer) {
+    dependsOn buildDebImage
+    targetImageId { buildDebImage.getImageId() }
+    portBindings = ['10037:18038', '8080:18088']
+}
+
+task createRpmContainer(type: DockerCreateContainer) {
+    dependsOn buildRpmImage
+    targetImageId { buildRpmImage.getImageId() }
+    portBindings = ['10037:18038', '8080:18088']
+}
+
+task startDebContainer(type: DockerStartContainer) {
+    dependsOn createDebContainer
+    targetContainerId { createDebContainer.getContainerId() }
+}
+
+task startRpmContainer(type: DockerStartContainer) {
+    dependsOn createRpmContainer
+    targetContainerId { createRpmContainer.getContainerId() }
+}
+
+task stopDebContainer(type: DockerStopContainer) {
+    dependsOn startDebContainer
+    targetContainerId { startDebContainer.getContainerId() }
+}
+
+task stopRpmContainer(type: DockerStopContainer) {
+    dependsOn startRpmContainer
+    targetContainerId { startRpmContainer.getContainerId() }
+}
+
+task removeDebContainer(type: DockerRemoveContainer) {
+    dependsOn stopDebContainer
+    targetContainerId { stopDebContainer.getContainerId() }
+}
+
+task removeRpmContainer(type: DockerRemoveContainer) {
+    dependsOn stopRpmContainer
+    targetContainerId { stopRpmContainer.getContainerId() }
+}
+
+task removeDebImage(type: DockerRemoveImage) {
+    dependsOn buildDebImage
+    targetImageId { buildDebImage.getImageId() }
+}
+
+task removeRpmImage(type: DockerRemoveImage) {
+    dependsOn buildRpmImage
+    targetImageId { buildRpmImage.getImageId() }
+}
+
+task dockerSmokeTest {
+    description = 'tests if the Repose instance can handle basic requests using Docker'
+    group = 'release'
+    dependsOn = ['buildDebImage',
+                 'buildRpmImage']
+    finalizedBy = ['removeDebImage',
+                   'removeRpmImage']
+}
+
+// Note: At this time, release verification will be handled out-of-band with the release. Therefore,
```

Comment:
I took a different approach this time (inspired by the Gradle Docker plugin's examples). Basically, I chose to leave the individual tasks open so that a user can get in-between them if so desired. That is, a user could just build the image, or build the image and a container, or what have you. Wanting to keep the built image around seems like it might be a common use-case (it certainly has been for the testing I've done) so it wouldn't make sense to immediately destroy the image right after creating it.

I did, however, ensure that the smoke test task cleans up after itself.